Refactor EVM withdraw tests to reduce duplication.

cwarnerdev opened the pull request:
Consolidated logic for testing withdraw functionality with failed transactions into a single helper function, `testEVMWithdrawWithFailedTx`. This reduces redundancy and makes it easier to test scenarios with different execution orderings.

jorgemmsilva commented on packages/vm/vmcontext/runreq.go:
```diff
@@ -80,6 +84,24 @@ func (vmctx *VMContext) creditAssetsToChain() {
+	case adjustmentOfTheCommonAccount < 0:
+		err := util.CatchPanicReturnError(func() {
+			vmctx.debitFromAccount(commonaccount.Get(vmctx.ChainID()), &iscp.Assets{
+				Iotas: uint64(-adjustmentOfTheCommonAccount),
```
right now I'm considering a positive number will be passed to `debitFromAccount`, but I can use absolute values or reconsider this

lunfardo314 replied:
I think it is fine what we have now, always have `uint64`. The adjustment can be negative and this is the only place in the code where we have to consider it.

jorgemmsilva commented on packages/vm/vmcontext/vmtxbuilder/totals.go:
```diff
@@ -0,0 +1,113 @@
+			ret.TokenBalances[nt.ID] = s
+		}
+	}
+	return
+}
+
+// sumOutputs sums all balances in outputs
+func (txb *AnchorTransactionBuilder) sumOutputs() (ret TransactionTotals) {
+	ret.TokenBalances = make(map[iotago.NativeTokenID]*big.Int)
+	ret.TotalIotas = txb.currentBalanceIotasOnAnchor
+
+	// sum up all initial values of internal accounts
+	for id, ntb := range txb.balanceNativeTokens {
+		if !ntb.producesOutput() {
+			continue
+		}
+		s, ok := ret.TokenBalances[id]
+		if !ok {
+			s = new(big.Int)
+		}
+		s.Add(s, ntb.balance)
+		ret.TokenBalances[id] = s
+		// sum up dust deposit in outputs of internal UTXOs
+		ret.TotalIotas += txb.vByteCostOfNativeTokenBalance()
+		ret.InternalDustDeposit += txb.vByteCostOfNativeTokenBalance()
+	}
+	for _, o := range txb.postedOutputs {
+		assets := assetsFromOutput(o)
+		ret.TotalIotas += assets.Iotas
+		for _, nt := range assets.Tokens {
+			s, ok := ret.TokenBalances[nt.ID]
+			if !ok {
+				s = new(big.Int)
+			}
+			s.Add(s, nt.Amount)
+			ret.TokenBalances[nt.ID] = s
+		}
+	}
+	return
+}
```
it seems a good chunk of code is repeated here, would it make sense to extract to a single function?

lunfardo314 replied:
optimized the code according to your suggestion. I think readability suffers now. Because actually the codes chunks are not identical, there are subtle differences in 2 places. Anyway, tests pass :)

cwarnerdev committed:
Refactor EVM withdraw tests to reduce duplication.

Consolidated logic for testing withdraw functionality with failed transactions into a single helper function, `testEVMWithdrawWithFailedTx`. This reduces redundancy and makes it easier to test scenarios with different execution orderings.
